corehq/ex-submodules/casexml/apps/phone/data_providers/case/load_testing.py

```diff
@@ -19,19 +27,24 @@ def _map_id(id, count):
     return CaseSyncUpdate(case, update.sync_token, required_updates=update.required_updates)
 
 
-def get_xml_for_response(update, restore_state):
+def get_xml_for_response(
+    update: CaseSyncUpdate,
+    restore_state: RestoreState,
+    total_cases: int,
+) -> list[str]:
```

Comment:
Can you explain why these annotations are being added here?

Reply:
Those annotations were actually added [in February](https://github.com/dimagi/commcare-hq/pull/31095) (that's how old the original PR is!) prior to our discussions about type hints. They come from before I dropped the type hints from [`livequery.compile_response()`](https://github.com/dimagi/commcare-hq/pull/31095/commits/5a0e99c4fc4bafc2921cb1eff45461025365c9e1). (This work partially motivated those discussions, in the hope that we could get some clarity on how best to use type hints in HQ.)

To answer your question, at the time, I used them so that I had some visibility into exactly what an "update" is, and left them so that the next person might also benefit.

I'll drop them.

But, as you know, I don't think that's a great option. In fact, I don't think _any_ of the options are _great_ options:

1. Leave this as is. Not verified by GitHub Actions. At least verified by IDEs. Helps the next dev unfamiliar with this area of the codebase.

2. Replace this with a docstring:
```python
def get_xml_for_response(update, restore_state, total_cases):
    """
    :param update: An instance of CaseSyncUpdate
    :param restore_state: An instance of RestoreState
    :param total_cases: An integer
    :returns: A list (not a string) of case XML (strings)
    """
```
That docstring is clumsy. It isn't verified by anything, not even an IDE. Not an improvement (imho).

3. Drop entirely:
```python
def get_xml_for_response(update, restore_state, total_cases):
```
The least helpful option. But it is nice and neat. And everyone is OK with this option. So, not perfect, but it'll do.

4. Just leave the useful bits:
```python
def get_xml_for_response(update: CaseSyncUpdate, restore_state, total_cases) -> list[str]:
```
An unholy violation of everything wholesome and beautiful. But useful. Allegedly "practicality beats purity", but how far does that stretch?

Reply:
There's another option, which is self documenting, carries through to the rest of the code, and is not controversial:

```py
def get_xml_for_response(case_sync_update, ...):
```